### PR TITLE
Parse security_opts before sending them to docker daemon

### LINKED ISSUE #6720: Docker driver security_opts field doesn't match flag behavior
Follow up from https://github.com/hashicorp/nomad/issues/5800, where it was shown that the `config.security_opt` stanza doesn't match the behavior of Docker when a file path is passed in as a `seccomp` profile.

Our documentation for the Docker task driver's `config.security_opt` stanza says:

> security_opt - (Optional) A list of string flags to pass directly to --security-opt. For example:
> ```
> config {
>  security_opt = [
>     "credentialspec=file://gmsaUser.json",
>   ]
> }
> ```

Because we're using the Docker HTTP API and not the Docker CLI to start Docker tasks, if the Docker CLI does any parsing of the command line arguments before passing them to the HTTP API, we'll see a difference in behavior in Nomad. It appears that this is the case for `security_opts` for the `seccomp` option in particular.

We can document this as a difference, but should we consider updating the task driver to match the Docker behavior to avoid this gotcha?

### PR DESCRIPTION
**Pull request by benbuzbee**
Fixes #6720

Copy the parsing function from the docker CLI. Docker daemon expects to see JSON for seccomp file not a path.

### CODE REVIEW

File: drivers/docker/driver_test.go [hunk 1]
```diff
@@ -1004,6 +1004,34 @@ func TestDockerDriver_SecurityOpt(t *testing.T) {
...
+	require.NoError(t, task.EncodeConcreteDriverConfig(cfg))
+
+	client, d, handle, cleanup := dockerSetup(t, task)
+	defer cleanup()
+	require.NoError(t, d.WaitUntilStarted(task.ID, 5*time.Second))
+
+	container, err := client.InspectContainer(handle.containerID)
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
```
Comment (notnoop) [inline drivers/docker/driver_test.go]:
```suggestion
require.NoError(t, err)
```

File: drivers/docker/driver_test.go [hunk 2]
```diff
@@ -1004,6 +1004,34 @@ func TestDockerDriver_SecurityOpt(t *testing.T) {
 	require.Exactly(t, cfg.SecurityOpt, container.HostConfig.SecurityOpt)
 }
 
+func TestDockerDriver_SecurityOptFromFile(t *testing.T) {
+	seccompPath := "./test-resources/docker/seccomp.json"
```
Comment (notnoop) [inline drivers/docker/driver_test.go]: I'd move this down closer to where it's used and keep the method preamble for test environment conditionals.

File: drivers/docker/driver_test.go [hunk 3]
```diff
@@ -1004,6 +1004,34 @@ func TestDockerDriver_SecurityOpt(t *testing.T) {
...
+
+	if runtime.GOOS == "windows" {
```
Comment (notnoop) [inline drivers/docker/driver_test.go]: You can also move the test to `driver_unix_test.go` (or `driver_linux_test.go`) instead if it doesn't apply to Windows.  Does it apply to macOS?

Comment (benbuzbee) [reply drivers/docker/driver_test.go]: Honestly not sure. I copied the boiler plate from the existing seccomp=unconfined test. I would have thought even windows would run the container in a VM where seccomp applies but I have no good way to test.

Comment (notnoop) [reply drivers/docker/driver_test.go]: Makes sense - I'll test behavior!

**Comment by notnoop** [comment]
Thank you so much @benbuzbee - this fix should be out in 0.11.0!

**Pull request merged**